### ISSUE TITLE
Provide an alternative to using subprocess.run patch

### ISSUE THREAD

**Issue by MaxOfLondon** [adamchainz/patchy#479]
### Description

It would be nice to have a fallback to `patch` if subprocess is not supported on the platform like for example `emscripten`.
An alternative could be https://github.com/conan-io/python-patch-ng.git but I am sure other also exist and could be evaluated.
Thank you in advance!

**Comment by adamchainz** [comment]
Yeah, I've always wanted to use a pure Python in-memory patch library. When I wrote patchy, none seemed easy. Perhaps python-patch-ng is the one, idk. If you'd like to put together a proof-of-concept that would be neat.
